packages/rum-core/src/browser/cookieObservable.ts

```diff
@@ -0,0 +1,64 @@
+import type { Configuration, CookieStore } from '@datadog/browser-core'
+import {
+  setInterval,
+  clearInterval,
+  Observable,
+  addEventListener,
+  ONE_SECOND,
+  findCommaSeparatedValue,
+} from '@datadog/browser-core'
+
+export interface CookieStoreWindow extends Window {
+  cookieStore?: CookieStore
+}
+
+export type CookieObservable = ReturnType<typeof createCookieObservable>
+
+export type CookieChange = { name: string; value: string | undefined }
+
+export function createCookieObservable(configuration: Configuration, cookieName: string) {
+  return new Observable<CookieChange>(
+    (observable) =>
+      listenToCookieStoreChange(configuration, cookieName, (event) => observable.notify(event)) ??
+      watchCookieFallback(cookieName, (event) => observable.notify(event))
+  )
+}
+
+function listenToCookieStoreChange(
+  configuration: Configuration,
+  cookieName: string,
+  callback: (event: CookieChange) => void
+) {
+  if (!(window as CookieStoreWindow).cookieStore) {
+    return
+  }
+
+  const listener = addEventListener(configuration, (window as CookieStoreWindow).cookieStore!, 'change', (event) => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏use existing constant
```suggestion
  const listener = addEventListener(configuration, (window as CookieStoreWindow).cookieStore!, DOM_EVENT.CHANGE, (event) => {
```